controller/controlplane/manager_options.go

```diff
@@ -537,11 +537,48 @@ func WithWatchNamespaces(watchNamespaces []string) managercfg.Opt {
 	}
 }
 
-// WithKonnectConfig sets the Konnect configuration for the manager.
-func WithKonnectConfig(konnectConfig *managercfg.KonnectConfig) managercfg.Opt {
+// WithKonnectOptions merges Konnect options from ControlPlane spec with the existing Konnect configuration.
+func WithKonnectOptions(konnectOptions *operatorv2alpha1.ControlPlaneKonnectOptions, existingKonnectConfig *managercfg.KonnectConfig) managercfg.Opt {
 	return func(c *managercfg.Config) {
-		if konnectConfig != nil {
-			c.Konnect = *konnectConfig
+		// Start with existing config if provided
+		if existingKonnectConfig != nil {
+			c.Konnect = *existingKonnectConfig
+		}
+
+		// Apply ControlPlane Konnect options if specified
+		if konnectOptions == nil {
+			return
+		}
+
+		// Configure consumer synchronization
+		if konnectOptions.ConsumersSync != nil {
```

Comment:
Is there any guarantee that the `c.Konnect` cannot be `nil` before calling the option? If not, I think we should check and set defaults. If there is such guarantee, a comment to explain it can reduce the confusion.